NDTensors/src/lib/GradedAxes/src/fusion.jl

```diff
@@ -18,26 +20,39 @@ function tensor_product(
   return foldl(tensor_product, (a1, a2, a3, a_rest...))
 end
 
-function tensor_product(a1::AbstractUnitRange, a2::AbstractUnitRange)
+function tensor_product(::AbstractUnitRange, ::AbstractUnitRange)
   return error("Not implemented yet.")
 end
 
 function tensor_product(a1::Base.OneTo, a2::Base.OneTo)
   return Base.OneTo(length(a1) * length(a2))
 end
 
-function tensor_product(a1::OneToOne, a2::AbstractUnitRange)
+function tensor_product(::OneToOne, a2::AbstractUnitRange)
   return a2
 end
 
-function tensor_product(a1::AbstractUnitRange, a2::OneToOne)
+function tensor_product(a1::AbstractUnitRange, ::OneToOne)
   return a1
 end
 
-function tensor_product(a1::OneToOne, a2::OneToOne)
+function tensor_product(::OneToOne, ::OneToOne)
   return OneToOne()
 end
 
+# Handle dual. Always return a non-dual GradedUnitRange.
+function tensor_product(a1::AbstractUnitRange, a2::UnitRangeDual)
+  return tensor_product(a1, flip(a2))
+end
+
+function tensor_product(a1::UnitRangeDual, a2::AbstractUnitRange)
+  return tensor_product(flip(a1), a2)
+end
+
+function tensor_product(a1::UnitRangeDual, a2::UnitRangeDual)
+  return tensor_product(flip(a1), flip(a2))
+end
```

Comment:
I think we discussed this, but I was picturing that tensor producting two dual spaces would result in a dual space, and also that users could select whether the output is dual or not (that's a feature we support in a similar function in ITensors.jl).

I'm ok if this is just a temporary definition and we reevaluate later, just wanted to know your thought process on this.

Reply:
Also if we decide to address that later, can you add a comment saying that we may reassess whether or not the output is dual, and also allow users to specify whether the output is dual or not?

Reply:
Deciding whether or not to output a dual space is a pure convention. I think it is more intuitive to always return a non-dual space, especially since `tensor_product(dual, nondual)` and  `tensor_product(nondual, dual)` already do. In generic code, it is hard to predict whether input spaces will be dual or not, with this convention the result can be predicted.

We already considered an optional argument `tensor_product(g1, g2, dual=false`), but we concluded that it was simpler to just write `dual(tensor_product(g1, g2)`. Therefore I would rather keep the current version.

Reply:
I think it is reasonable to define `tensor_product(dual, dual) -> dual`, we define it that way in ITensors.jl and it is a natural convention for functions like SVD, but we can put that decision off for now and see how it goes with the current definition. I agree one could always post-process the output (though maybe it should be `flip(tensor_product(g1, g2))` to output a dual space, since that preserves the space).

Reply:
Can you add a comment above this function saying that we might consider other conventions for the duality of the space in the future?